64_FFT/main.cpp

```diff
@@ -0,0 +1,326 @@
+	using asset_base_t = application_templates::MonoAssetManagerAndBuiltinResourceApplication;
+
+	// This is the first example that submits multiple workloads in-flight. 
+	// What the shader does is it computes the minimum distance of each point against K other random input points.
+	// Having the GPU randomly access parts of the buffer requires it to be DEVICE_LOCAL for performance.
+	// Then the CPU downloads the results and finds the median minimum distance via quick-select.
+	// This bizzare synthetic workload was specifically chosen for its unfriendliness towards simple buffer usage.
+	// The fact we have variable sized workloads and run them in a loop means we either have to dynamically
+	// suballocate from a single buffer or have K worst-case sized buffers we round robin for K-workloads in flight.
+	// Creating and destroying buffers at runtime is not an option as those are very expensive operations. 
+	// Also since CPU needs to heapify the outputs, we need to have the GPU write them into RAM not VRAM.
+	smart_refctd_ptr<IGPUComputePipeline> m_pipeline;
+
+	// The Utility class has lots of methods to handle staging without relying on ReBAR or EXT_host_image_copy as well as more complex methods we'll cover later.
+	// Until EXT_host_image_copy becomes ubiquitous across all Nabla Core Profile devices, you need to stage image copies from an IGPUBuffer to an IGPUImage.
+	// Why use Staging for buffers in the age of ReBAR? While GPU workloads overlap the CPU, individual GPU workloads's execution might not overlap each other
+	// but their data might. In this case you want to "precisely" time the data update on the GPU timeline between the end and start of a workload.
+	// For very small updates you could use the commandbuffer updateBuffer method, but it has a size limit and the data enqueued takes up space in the commandpool.
+	// Sometimes it might be unfeasible to either have multiple copies or update references to those copies without a cascade update.
+	// One example is the transformation graph of nodes in a scene, where a copy-on-write of a node would require the update the offset/pointer held by
+	// any other node that refers to it. This quickly turns into a cascade that would force you to basically create a full copy of the entire data structure
+	// after most updates. Whereas with staging you'd "queue up" the much smaller set of updates to apply between each computation step which uses the graph.
+	// Another example are UBO and SSBO bindings, where once you run out of dynamic bindings, you can no longer easily change offsets without introducting extra indirection in shaders.
+	// Actually staging can help you re-use a commandbuffer because you don't need to re-record it if you don't need to change the offsets at which you bind!
+	// Finally ReBAR is a precious resource, my 8GB RTX 3070 only reports a 214MB Heap backing HOST_VISIBLE and DEVICE_LOCAL device local memory type.
+	smart_refctd_ptr<nbl::video::IUtilities> m_utils;
+
```

Comment:
try to remove the comments (and dead code) that were from the example you originally forked, we don't want to be repeating ourselves